lib/transformer/dataset_cast.rb

```diff
@@ -1,17 +1,31 @@
 module Transformer
   class DatasetCast
     include Virtus.model
-    include Households
-    include Buildings
-    include Transport
-    include Industry
-    include Agriculture
-    include Supply
-    include Other
+    include ActiveModel::Validations
 
-    attribute :number_of_inhabitants,              Integer, mandatory: true, unit: ''
-    attribute :number_of_residences,               Float, mandatory: true, unit: ''
-    attribute :number_of_residences_with_solar_pv, Integer, mandatory: true, unit: ''
-    attribute :percentage_of_old_residences,       Float, mandatory: true, unit: '%'
+    attribute :has_industry, Boolean, default: false
+    attribute :has_agriculture, Boolean, default: false
+    attribute :percentage_of_old_residences, Float, default: 0.0
+
+    validates_presence_of :number_of_residences
+    validates :number_of_residences, numericality: { greater_than: 0 }
+
+    def initialize(_ = {}, *args)
+      extend(Virtus.model)
```

Comment:
The `Atlas::Dataset` attributes can be moved to compile time, but the `GraphMethods` can't somehow since the `data_dir` of `Atlas` is not known at that moment in the code. I could move that up in the execution and see if that makes sense.

Reply:
>  since the data_dir of Atlas is not known at that moment in the code

Ah, that's annoying, but it makes sense: if a new version of ETSource is loaded in ETLocal the graph methods might change. 😬 

Perhaps we could use a `graph_methods` attribute to store all of them together?

```ruby
require 'ostruct'

class DatasetCast
  attribute :graph_methods, OpenStruct, default: ->(*) { OpenStruct.new }
end

dc = DatasetCast.new

dc.graph_methods.x_demand = 5.0
dc.graph_methods.y_number_of_units = 10.0

dc.graph_methods.y_number_of_units
# => 10.0

dc.graph_methods.to_h
# => { :x_demand => 5.0, :y_number_of_units => 10.0 }
```

Reply:
`OpenStruct` is quite fluid but might actually be useful in this situation 👼 will play around with it, to see if it fits. Statically defining all those attributes also seems way to much. 😅